### PR TITLE
(SIMP-4663) Use systemd for tmp.mounts if possible

### PR DESCRIPTION
**Pull request by trevor-vaughan**
- Refactor the simp::mountpoints::tmp to use systemd's tmp.mount target if
  the system supports systemd.

SIMP-4663 #close

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
+* Thu Apr 12 2019 Trevor Vaughan <tvaughan@onyxpoint.com> - 4.8.0-0
+- Refactor the simp::mountpoints::tmp to use systemd's tmp.mount target if
+  the system supports systemd.
 - Fixed a bug where the root password field was attempting to set an 'undef'
   value as Sensitive.
 - Bumped the supported Puppet version to include Puppet 6
 - Removed Puppet 4 from the supported list
 
-* Mon Mar 25 2019 Joseph Sharkey <shark.bruhaha@gmail.com> - 4.8.0-0
+* Tue Apr 09 2019 Joseph Sharkey <shark.bruhaha@gmail.com> - 4.8.0-0
```
Comment (sharkbruhaha) [inline CHANGELOG]: whoops, thanks for that catch

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-simp/pipelines/56552234

**Comment by trevor-vaughan** [comment]
Looks like I need to prod the acceptance tests a bit and figure out why the remounts are happening. Looks like it may be an SELinux twiddle that needs to happen.

**Comment by sharkbruhaha** [comment]
@trevor-vaughan  I've proded the tests a couple times now and they keep failing on a number of acceptance tests: https://gitlab.com/simp/pupmod-simp-simp/pipelines/56552809

**Comment by trevor-vaughan** [comment]
Rebased

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-simp/pipelines/57596617

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-simp/pipelines/57615576

**Comment by sharkbruhaha** [comment]
https://gitlab.com/simp/pupmod-simp-simp/pipelines/57618435

**Comment by trevor-vaughan** [comment]
:-| I have no idea why the `tmp.mounts` service is failing. This may be an issue in the underlying provider.

I'll double check it locally (again).

**Comment by sharkbruhaha** [comment]
Was just about to msg you that I had prodded them to no avail.

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-simp/pipelines/57850799

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-simp/pipelines/57854516

**Pull request merged**